vulkano/src/command_buffer/synced/base.rs

```diff
@@ -636,7 +637,36 @@ impl<P> SyncCommandBufferBuilder<P> {
                     // Indicate to the image that a memory barrier has been inserted that
                     // transitions it out of an undefined state.
                     unsafe {
-                        img.layout_initialized();
+                        if !img.is_layout_initialized() {
+                            let init_layout = if img.preinitialized_layout() {
+                                ImageLayout::Preinitialized
+                            } else {
+                                ImageLayout::Undefined
+                            };
+                            if initial_layout_requirement != init_layout {
+                                let mut b = UnsafeCommandBufferBuilderPipelineBarrier::new();
+                                b.add_image_memory_barrier(img,
+                                                            0 .. img.mipmap_levels(),
+                                                            0 .. img.dimensions().array_layers(),
+                                                            PipelineStages {
+                                                                bottom_of_pipe: true,
+                                                                ..PipelineStages::none()
+                                                            },
+                                                            AccessFlagBits::none(),
+                                                            stages,
+                                                            access,
+                                                            true,
+                                                            None,
+                                                            if img.preinitialized_layout() {
+                                                                ImageLayout::Preinitialized
+                                                            } else {
+                                                                ImageLayout::Undefined
+                                                            },
```

Comment:
Could use the `init_layout` variable here rather than checking `img.preinitialized_layout()` again